app/controllers/search_controller.rb

```diff
@@ -30,7 +30,10 @@ def extract_errors(response)
   end
 
   def extract_filters(response)
-    response&.data&.search&.to_h&.dig('aggregations')
+    aggs = response&.data&.search&.to_h&.dig('aggregations')
+    return if aggs.blank?
+
+    aggs.select { |agg_key, agg_values| agg_values.present? }
```

Comment:
If this is an actually possible state to enter, then yes this is a great place to do it. I guess I just didn't understand how this would happen. Thanks!

Reply:
Yeah, it happens whenever there are no results, e.g.: https://search.libraries.mit.edu/results?q=asdofyuasd+fuygasdfads&advanced=&citation=&contributors=&fundingInformation=&identifiers=&locations=&subjects=&title=

Reply:
Doh! I thought the line `return if aggs.blank?`. I now understand this, thanks for talking me through it.

Reply:
Ah! The guard clause actually covers a different weird edge case. :) If the API returns errors, there are no results/aggregations in the response. In that case, `aggs` would be `nil`, and we'd be trying to call `select` on `nil`.

Another solution I considered was to add `if @errors.nil?` to the `extract_filters` method call, [as we do with pagination](https://github.com/MITLibraries/timdex-ui/blob/d528e9f6a8ab57d149b1241f5b692826b6632bb2/app/controllers/search_controller.rb#L19). If you think that's more readable, I can drop the guard clause and go that route instead.